drivers/SmartThings/matter-button/src/init.lua

```diff
@@ -3,172 +3,222 @@ local log = require "log"
-  local MSR = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.SwitchFeature.MOMENTARY_SWITCH_RELEASE})
-  local MSL = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.SwitchFeature.MOMENTARY_SWITCH_LONG_PRESS})
-  local MSM = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.SwitchFeature.MOMENTARY_SWITCH_MULTI_PRESS})
+  if device.network_type ~= device_lib.NETWORK_TYPE_CHILD then
+    local MS = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.SwitchFeature.MOMENTARY_SWITCH})
+    -- local LS = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.SwitchFeature.LATCHING_SWITCH})
 
-  if #MS > 0 then
-    if #MSL < 1 and #MSM < 1 then
-      device:set_field(BUTTON_TYPE, BUTTON_TYPE_MAP.SINGLE_PRESS) --singlepress
-      device:emit_event(capabilities.button.supportedButtonValues({"pushed"}, {visibility = {displayed = false}})) --supported values
+    -- find the default/main endpoint, the device with the lowest EP that supports MS
+    table.sort(MS)
+    local main_endpoint = MS[1] -- the endpoint matching to the non-child device
+    if MS[1] == 0 then main_endpoint = MS[2] end -- we shouldn't hit this, but just in case
 
-    elseif #MSR > 0 and #MSL > 0 and #MSM < 1 then
-      device:set_field(BUTTON_TYPE, BUTTON_TYPE_MAP.SINGLE_PRESS) --singlepress
-      device:emit_event(capabilities.button.supportedButtonValues({"pushed", "held"},{visibility = {displayed = false}})) --supported values
+    local MSR = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.SwitchFeature.MOMENTARY_SWITCH_RELEASE})
+    local MSL = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.SwitchFeature.MOMENTARY_SWITCH_LONG_PRESS})
+    local MSM = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.SwitchFeature.MOMENTARY_SWITCH_MULTI_PRESS})
 
-    elseif #MSM > 0 then
-      device:set_field(BUTTON_TYPE, BUTTON_TYPE_MAP.MULTI_PRESS) -- multipress
+    -- We have a static profile that will work for this number of buttons
+    if contains(STATIC_PROFILE_SUPPORTED, #MS) then
+      device:try_update_metadata({profile = string.format("%d-button", #MS)})
+    end
 
-      --send read MultiPressMax attribute request
-      local req = clusters.Switch.attributes.MultiPressMax:read(device)
-      device:send(req)
+    -- At the moment, we're taking it for granted that all momentary switches only have 2 positions
+    -- TODO: flesh this out for NumberOfPositions > 2
+    local current_component_number = 2
+    for _, ep in ipairs(MS) do -- for each momentary switch endpoint (including main)
+      -- build the mapping of endpoints to components if we have a static profile (multi-component)
+      if contains(STATIC_PROFILE_SUPPORTED, #MS) then
+        local map = device:get_field(ENDPOINT_TO_COMPONENT_MAP) or {}
```

Comment:
that's only true for the _first_ endpoint, but this loops through all of the momentary switch endpoints, so by the second iteration it will be non-null

Reply:
🤦🏼 Youre right.